transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,251 @@
+        if seasonal_service_start is None
+        else "Out of Season"
+    )
+
+
+def get_service_code_exemption_map(
+    organisation_id: int,
+) -> Dict[str, ServiceCodeExemption]:
+    return {
+        service.registration_number.replace("/", ":"): service
+        for service in ServiceCodeExemption.objects.add_registration_number().filter(
+            licence__organisation__id=organisation_id
         )
+    }
 
-    def get_queryset(self):
-        return OTCService.objects.get_all_without_exempted_ones(self._organisation_id)
 
-    def to_csv(self):
-        string_ = io.StringIO()
-        writer = csv.writer(string_, quoting=csv.QUOTE_ALL)
-        writer.writerow(
-            ("Service Status", "Licence Number", "Service Code", "Line", "Dataset ID")
+def get_all_otc_map(organisation_id: int) -> Dict[str, OTCService]:
+    """
+    Get a dictionary which includes all OTC Services for an organisation.
+    """
+    return {
+        service.registration_number.replace("/", ":"): service
+        for service in OTCService.objects.get_all_otc_data_for_organisation(
+            organisation_id
         )
+    }
+
+
+def get_seasonal_service_map(organisation_id: int) -> Dict[str, SeasonalService]:
+    """
+    Get a dictionary which includes all the Seasonal Services
+    for an organisation.
+    """
+    return {
+        service.registration_number.replace("/", ":"): service
+        for service in SeasonalService.objects.filter(
+            licence__organisation_id=organisation_id
+        )
+        .add_registration_number()
+        .add_seasonal_status()
+    }
+
 
-        for otc_service in self.get_queryset():
-            row = self.get_row(otc_service)
-            if row[0] == self.MISSING_DATA:
-                writer.writerow(row)
-
-        for otc_service in self.get_queryset():
-            row = self.get_row(otc_service)
-            if row[0] == self.REGISTERED:
-                writer.writerow(row)
-
-        string_.seek(0)
-        return string_
-
-    def get_row(self, otc_service: OTCService) -> Tuple:
-        datasets: Set = {
-            str(dataset_id)
-            for service_code, dataset_id in self.org_service_code_to_dataset
-            if service_code == otc_service.service_code
-        }
-        return (
-            self.REGISTERED if len(datasets) > 0 else self.MISSING_DATA,
-            otc_service.licence.number,
-            otc_service.service_code,
-            otc_service.service_number,
-            ";".join(datasets),
+class ServiceCodesCSV(CSVBuilder):
+    columns = [
+        CSVColumn(
+            header="Requires Attention",
+            accessor=lambda otc_service: otc_service.get("require_attention"),
+        ),
+        CSVColumn(
+            header="Published Status",
+            accessor=lambda otc_service: "Published"
+            if otc_service.get("dataset_id")
+            else "Unpublished",
+        ),
+        CSVColumn(
+            header="OTC status",
```

Comment:
I'm sure they would rather this was capitalised, since all the other statuses are.

```suggestion
            header="OTC Status",
```